src/test/java/org/numenta/nupic/encoders/RandomDistributedScalarEncoderTest.java

```diff
@@ -0,0 +1,438 @@
+ * 
+ * @author Anubhav Chaturvedi
+ *
+ */
+public class RandomDistributedScalarEncoderTest {
+
+	private RandomDistributedScalarEncoder rdse;
+	private RandomDistributedScalarEncoder.Builder builder;
+	
+	@Rule
+	public ExpectedException exception = ExpectedException.none();
+	
+	void setUp()
+	{
+		builder =  RandomDistributedScalarEncoder.builder()
+			        .n(500)
+			        .w(23)
+			        .resolution(1)
+			        .setOffset(0);
+	}
+	
+	@Test
+	public void testEncoding()
+	{
+		builder = RandomDistributedScalarEncoder.builder()
+				.name("enc")
+				.resolution(1)
+				.w(23)
+				.n(500)
+				.setOffset(0);
+		rdse = builder.build();
+		
+		
+		int e0[] = rdse.encode(-0.1);
+		assertEquals("Number of on bits is incorrect", getOnBits(e0), 23);
+		assertEquals("Width of the vector is incorrect", e0.length, 500);
+		assertEquals("Offset doesn't correspond to middle bucket", rdse.getBucketIndices(0)[0], 
+				rdse.getMaxBuckets()/2);
+		assertEquals("Number of buckets is not 1", 1, rdse.bucketMap.size());
+		
+		
+		int e1[] = rdse.encode(1.0);
+		assertEquals("Number of buckets is not 2", 2, rdse.bucketMap.size());
+		assertEquals("Number of on bits is incorrect", getOnBits(e1), 23);
+		assertEquals("Width of the vector is incorrect", e0.length, 500);
+		assertEquals("Overlap is not equal to w-1", computeOverlap(e0, e1), 22);
+		
+		
+		int e25[] = rdse.encode(25.0);
+		assertTrue("Buckets created are not more than 23", rdse.bucketMap.size()>23);
+		assertEquals("Number of on bits is incorrect", getOnBits(e1), 23);
+		assertEquals("Width of the vector is incorrect", e0.length, 500);
+		assertTrue("Overlap is too high", computeOverlap(e0, e25) < 4);
+		
+		
+		assertArrayEquals("Encodings are not consistent - they have changed after new buckets "
+				+ "have been created", rdse.encode(-0.1), e0);
+		assertArrayEquals("Encodings are not consistent - they have changed after new buckets "
+				+ "have been created", rdse.encode(1.0), e1);
+	}
+	
+	@Test
+	public void testMissingValues()
+	{
+		builder = RandomDistributedScalarEncoder.builder()
+				.name("enc")
+				.resolution(1);
+		rdse = builder.build();
+		
+		int[] e2 = rdse.encode(Double.NaN);
+		assertEquals(0, getOnBits(e2));
+		
+		int[] e1 = rdse.encode(Encoder.SENTINEL_VALUE_FOR_MISSING_DATA);
+		assertEquals(0, getOnBits(e1));
+	}
+	
+	@Test
+	public void testResolution()
+	{
+		builder = RandomDistributedScalarEncoder.builder()
+				.name("enc")
+				.resolution(1);
+		rdse = builder.build();
+		
+		int[] e23 = rdse.encode(23.0);
+		int[] e23_1 = rdse.encode(23.1);
+		int[] e22_9 = rdse.encode(22.9);
+		int[] e24 = rdse.encode(24.0);
+		
+		assertEquals(rdse.getW(), getOnBits(e23));
+		assertArrayEquals("Numbers within resolution don't have the same encoding", e23_1, e23);
+		assertArrayEquals("Numbers within resolution don't have the same encoding", e22_9, e23);
+		
+		assertFalse("Numbers outside resolution have the same encoding", Arrays.equals(e23, e24));
```

Comment:
Sure.

``assertFalse(x)`` can only check the value of the boolean. When the assertion fails it will just report (with the message of course) "expected false".

``assertThat(a, equalTo(b))`` is smarter and knows that it's an array equality assertion. When it fails it'll report the two arrays for comparison.

This is different than ``assertEquals(a, b)`` as that uses ``==`` under the hood and two arrays with the same values don't pass that comparison.


Just a suggestion.